### PR TITLE
BUG: special: Ensure zetac(1) returns inf.

### PR DESCRIPTION
**Pull request by WarrenWeckesser**
Fixes gh-8335.

### CODE REVIEW

**Review by tylerjereddy — COMMENTED** [review]
I'm ok with the C level changes made in this PR. However, I do have to point out that the [zetac docs](https://docs.scipy.org/doc/scipy/reference/generated/scipy.special.zetac.html#scipy.special.zetac) indicate that the function is defined for `x > 1`, where `x` is also used as the input parameter for `zetac`.

So, while `infinity - 1 = infinity` seems to make perfect sense, the docstring might lead one to believe that i.e., an exception might be raised if `x <= 1`.

Should that be clarified too? I'm not a domain expert on the zeta function, but maybe a docstring adjustment is in order too if this fix is justified.

**Comment by WarrenWeckesser** [comment]
@tylerjereddy Good point.  In fact, `zetac` implements the extension of the zeta function to the real line:

```
In [3]: zetac([-1, -0.5, 0, 0.5, 1, 1.5])
Out[3]: 
array([ -1.08333333e+00,  -1.20788622e+00,  -1.50000000e+00,
        -2.46035451e+00,   1.70141183e+38,   1.61237535e+00])
```
I'll add an update of the docstring to this pull request.

**Comment by WarrenWeckesser** [comment]
Huh.  `zetac(x)` returns 0 for `x < -30.8148`.   If the algorithm doesn't work for more negative values, it should raise an error.  I'm not going to dig into the `zetac` code to see if this can be fixed.  I'll document the current behavior when I update the docstring, and create a new issue for it.

**Comment by WarrenWeckesser** [comment]
Bug report for `zetac(x)` returning 0 for `x < -30.8148`: https://github.com/scipy/scipy/issues/8371

**Comment by person142** [comment]
>  I'm not going to dig into the zetac code to see if this can be fixed

Could you be convinced to change the return at

https://github.com/scipy/scipy/blob/master/scipy/special/cephes/zetac.c#L180

to NaN?

**Comment by WarrenWeckesser** [comment]
@person142 I guess it wouldn't have hurt to dig a *little*. 😄 

I changed the return value to `nan`, but I didn't add a unit test for it.  This feels like a temporary fix, so I'd rather not enshrine that behavior by blessing it with a test.

File: scipy/special/add_newdocs.py [hunk 1]
```diff
@@ -6811,12 +6811,42 @@ def add_newdoc(place, name, doc):
...
+    `nan` for `x` less than -30.8148.
+
+    Parameters
+    ----------
+    x : array_like of float
+        Values at which to compute zeta(x) - 1 (must be real).
+
+    Returns
+    -------
+    float
```
Comment (larsoner) [inline scipy/special/add_newdocs.py]: Array-like of float (to match input description, since it should be the same shape right?)

**Review by larsoner — APPROVED** [review]
Other than a minor doc gripe LGTM

File: scipy/special/add_newdocs.py (hunk 1 above)
Comment (WarrenWeckesser) [reply scipy/special/add_newdocs.py]: Sure.  There doesn't appear to be a consistent style for describing the return value of ufunc-like functions: `ndarray`, `array`, `float`, `array_like` are some of the variations.  (Personally I've never liked `array_like`.  Why use an underscore?  There is no such Python object, it is not an identifier, so why not use the normal dash, and call it "array-like"?  But I'll begrudgingly use it--choose your battles and all that.)

**Comment by person142** [comment]
Thanks @WarrenWeckesser.

**Pull request merged**